src/v/cloud_roles/refresh_credentials.h

```diff
@@ -181,4 +181,31 @@ struct endpoint {
 static constexpr retry_params default_retry_params{
   .backoff_ms = std::chrono::milliseconds{500}, .max_retries = 8};
 
+template<typename CredentialsProvider>
+refresh_credentials make_refresh_credentials(
+  ss::gate& gate,
+  ss::abort_source& as,
```

Comment:
I'm thinking about this again seeing this here:

I'm wondering if gate should not be passed in here, but rather the refresh_credentials should manage its own gate and expose something like a stop interface method.

If the refresh_credentials has a stop method, it can also avoid needing to take an abort source, too.

The benefit of this is that it makes the refresh_credentials more self-contained and easier to understand with respect to the lifetime of these reference, and reducing coupling. it might not be appropriate here, but it kind of looks like it would work ok.